addons/mail/static/src/models/notification_group/notification_group.js

```diff
@@ -67,6 +71,31 @@ function factory(dependencies) {
             return `${this.modelName}_${data.id}`;
         }
 
+        /**
+         * Compute the most recent date inside the notification messages.
+         *
+         * @private
+         * @returns {String|undefined}
+         */
+        _computeDate() {
+            const dates = this.notifications.filter((notification) => notification.message && notification.message.date).map(notification => notification.message.date);
+            if (dates.lenght == 0) {
+                return clear();
+            }
+            const groupDate = moment.max(dates);
+            return groupDate;
+        }
+
+        /**
+         * Compute the position of the group inside the notification list.
+         *
+         * @private
+         * @returns {int}
+         */
+        _computeSequence() {
+            return Math.min(...this.notifications.map(notification => notification.message.id));
```

Comment:
Lower sequence comes first, and we want most recent first? So shouldn't it use max instead (cf. comment in test)?